### ISSUE TITLE
Tenancy Token - Placeholder Search Privacy Issue

### ISSUE THREAD

**Issue by akilic942** [meilisearch/meilisearch#4602]
Hello there,

We have one index being shared by multiple users. Each User gets his own tenancy token, so they only can see their own data (the tenancy token adds a filter). While the tenancy token works, it gets ignored when the placeholder search occurs. So when the page gets loaded initially, it shows the data of the last user. 

Meilisearch Javascript SDK 0.38.0
Meilisearch Cloud Service 1.7.6 

**To Reproduce**
1. Log In to first user
2. Open page with table, see initially loading own data (or not)
3. Log out
4. Log in to the second user
5. Open page with table, see initially loading data of the last user
6. Reload page
7. See initially loading own data

**Expected behavior**
The filter of the tenancy token should be considered in the placeholder search. When the user does a placeholder search, he should not see any data, that should not pass the filters.

**Meilisearch version:**
Meilisearch Javascript SDK 0.38.0
Meilisearch Cloud Service 1.7.6 

**Additional context**
I added a security check where I check if there is any data that should not be in the results, if so, it should run the query again. It needs 3-8 queries until it gives the right data.

I'm pretty sure that this is a meilisearch issue, as this happens for each Index not for each meilisearchClient instance 

Tested on multiple devices and browsers.

**Comment by Kerollmops** [comment]
Hey @akilic942 👋

I just tried using cURL, and it works perfectly when I use the tenant token generated with [this online JSON Webtoken generator](https://jwt.io/). I used the default movies dataset you can find on the getting started page of the documentation and added a filter to only display _the Matrix_ movie (`id = 603`).

Would it be possible to tell me more about the way you configured your JS client? Or give me a reproducer with cURL requests, please?

Have a nice day 🌔

**Comment by Fronix** [comment]
@akilic942 It sounds like you have a searchtoken or apikey configured as default (in the JS SDK) and it's being used until the users token kicks in. We are using this type of setup and we make sure to fetch the users token before any search is made, if the users token isn't able to be fetched the whole search will just fail with unauthorized as it should.

This is easily achieved by using a custom `httpclient` or if you already have the token `custom request config`
https://github.com/meilisearch/meilisearch-js?tab=readme-ov-file#custom-request-config-
https://github.com/meilisearch/meilisearch-js?tab=readme-ov-file#custom-http-client-

**Comment by akilic942** [comment]
Thank you for you replies @Kerollmops @Fronix 

>I just tried using cURL, and it works perfectly when I use the tenant token generated 
I tried it myself, i get the same results as I stated in my first response. The first place holder search gives me the results of the last placeholder search. The second place holder search finally uses the right filters.

>Would it be possible to tell me more about the way you configured your JS client? Or give me a reproducer with cURL requests, please?

`curl --location 'https://edge.meilisearch.com/indexes/Customers/search' \
--header 'Authorization: Bearer XXXXXX' \
--data ''`

While i was typing this, i changed https://edge.meilisearch.com (Analytics and monitoring)  to the original meilisearch-cloud url - and voila, that fixed it.

**It seems to be a bug with edge.meilisearch.com Analytics and monitoring**
